forte/proc/dsrg.py

```diff
@@ -245,6 +245,26 @@ def compute_energy(self):
 
         return e_dsrg if len(self.energies) == 0 else e_relax
 
+    def compute_gradient(self, coupling_coefficients, ci_vectors):
+        """ Compute DSRG-MRPT2 analytic gradients. """
+        # Notes (Shuhe Wang):
+
+        e_relax = 0.0  # initialize this to avoid PyCharm warning
+
+        # Perform the initial un-relaxed DSRG
```

Comment:
A safeguard of this function: test the reference relaxation. If the energy is not relaxed, proceed. If not, raise an error.

Reply:
Well, when the gradient('forte') is called, the dertype is automatically set to "first" in pymodule.py. psi4.core.set_global_option('DERTYPE', 'FIRST'),
so I think we don't need to add that condition.

Reply:
Wait, I am talking about reference relaxation here.

I see your solution works now because inside this function there is `compute_energy` but I am trying to argue `compute_energy` should not live inside `compute_gradient`.

Reply:
solved